aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1285,6 +1285,25 @@ std::vector<Tensor> unsafe_split(const Tensor& self, int64_t split_size, int64_t
   return result;
 }
 
+std::vector<Tensor> hsplit(const Tensor& self, int64_t split_size) {
+  TORCH_CHECK(self.dim() >= 1, "torch.hsplit requires a tensor with at least 1 dimension")
+  int64_t dim = (self.dim() == 1) ? 0 : 1;
+  TORCH_CHECK(self.sizes()[dim] % split_size == 0, "self.sizes()[", dim, "]=", self.sizes()[dim], " is not divisible by split_size=", split_size);
```

Comment:
start error error messages with the name of the function "torch.hsplit" in the case of hsplit, like the first error message, and use the names in the documentation to refer to values. For example, in this case, use "input" instead of "self" to refer to the tensor